### PR TITLE
✨(frontend) remove team modal

### PR DESCRIPTION
**Pull request by AntoLC**
## Purpose

We can now delete our team.

## Proposal

- [x] add endpoint to remove team
- [x] integrate modal, controls and interactions
- [x] tests

## Demo

[scrnli_3_25_2024_5-42-22 PM.webm](https://github.com/numerique-gouv/people/assets/25994652/3c6038d1-6abe-408d-91c6-477a0962d2dd)

### CODE REVIEW

File: src/frontend/apps/desk/src/features/teams/components/TeamActions.tsx [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+              onClick={() => {
+                setIsModalRemoveOpen(true);
+                setIsDropOpen(false);
+              }}
+              color="primary-text"
+              icon={<span className="material-icons">delete</span>}
+            >
+              <Text $theme="primary">{t('Delete the team')}</Text>
+            </Button>
+          )}
```
Comment (lebaudantoine) [inline src/frontend/apps/desk/src/features/teams/components/TeamActions.tsx:61]: I think these translations are missing
Capture d’écran 2024-03-26 à 13 54 32

File: src/frontend/apps/e2e/__tests__/app-desk/common.ts [hunk 2]
```diff
@@ -46,3 +46,45 @@ export const createTeam = async (
...
+  // Select a new user
+  await inputSearch.fill(fillText);
+
+  // Intercept response
+  const responseSearchUser = await responsePromiseSearchUser;
+  const users = (await responseSearchUser.json()).results as {
+    name: string;
+  }[];
+
+  console.log('index' + index, users[index].name);
```
Comment (lebaudantoine) [inline src/frontend/apps/e2e/__tests__/app-desk/common.ts:73]: to be removed

File: src/frontend/apps/e2e/__tests__/app-desk/teams-delete.spec.ts [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+    await page.getByRole('button', { name: 'Validate' }).click();
+
+    // Delete the team button should be hidden
+    await page.getByLabel(`Open the team options`).click();
+    await expect(
+      page.getByRole('button', { name: `Delete the team` }),
+    ).toBeHidden();
+  });
+
+  test('it deletes the team when we are member', async ({
```
Comment (lebaudantoine) [inline src/frontend/apps/e2e/__tests__/app-desk/teams-delete.spec.ts:54]: nitpicking: `'it cannot delete the team when we are member'`

File: src/frontend/apps/e2e/__tests__/app-desk/teams-delete.spec.ts [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+    await page.getByLabel(`Open the team options`).click();
+    await page.getByRole('button', { name: `Delete the team` }).click();
+    await page.getByRole('button', { name: `Confirm deletion` }).click();
+    await expect(page.getByText(`The team has been removed.`)).toBeVisible();
+    await expect(
+      page.getByRole('button', { name: `Create a new team` }),
+    ).toBeVisible();
+  });
+
+  test('it deletes the team when we are admin', async ({
```
Comment (lebaudantoine) [inline src/frontend/apps/e2e/__tests__/app-desk/teams-delete.spec.ts:26]: nitpicking: `'it cannot delete the team when we are admin'`

**Review by lebaudantoine — COMMENTED** [review]
Everything looks good to me, two translations are missing and we good

File: src/frontend/apps/desk/src/features/teams/components/TeamActions.tsx (hunk 1 above)
Comment (AntoLC) [reply src/frontend/apps/desk/src/features/teams/components/TeamActions.tsx:61]: [fixup! 🛂(app-desk) add TeamActions component](https://github.com/numerique-gouv/people/pull/148/commits/da4456fb3ae86356220a8c96357d6c650d004a77)

File: src/frontend/apps/e2e/__tests__/app-desk/common.ts (hunk 2 above)
Comment (AntoLC) [reply src/frontend/apps/e2e/__tests__/app-desk/common.ts]: [fixup! 🛂(app-desk) add TeamActions component](https://github.com/numerique-gouv/people/pull/148/commits/da4456fb3ae86356220a8c96357d6c650d004a77)

File: src/frontend/apps/e2e/__tests__/app-desk/teams-delete.spec.ts (hunk 3 above)
Comment (AntoLC) [reply src/frontend/apps/e2e/__tests__/app-desk/teams-delete.spec.ts]: [fixup! 🛂(app-desk) add TeamActions component](https://github.com/numerique-gouv/people/pull/148/commits/da4456fb3ae86356220a8c96357d6c650d004a77)

File: src/frontend/apps/e2e/__tests__/app-desk/teams-delete.spec.ts (hunk 4 above)
Comment (AntoLC) [reply src/frontend/apps/e2e/__tests__/app-desk/teams-delete.spec.ts]: [fixup! 🛂(app-desk) add TeamActions component](https://github.com/numerique-gouv/people/pull/148/commits/da4456fb3ae86356220a8c96357d6c650d004a77)

**Review by lebaudantoine — APPROVED** [review]
LGTM

**Pull request merged**